.github/workflows/ci-instrumentation.yml

```diff
@@ -0,0 +1,131 @@
+name: CI Instrumentation
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - 'release/**'
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  instrumentation:
+    strategy:
+      fail-fast: false
+      matrix:
+        gem:
+          - action_pack
+          - action_view
+          - active_job
+          - active_model_serializers
+          - active_record
+          - active_support
+          - all
+          - aws_sdk
+          - base
+          - concurrent_ruby
+          - delayed_job
+          - ethon
+          - excon
+          - faraday
+          - graphql
+          - http
+          - http_client
+          - koala
+          - lmdb
+          - net_http
+          - rack
+          - rails
+          - restclient
+          - rspec
+          - sinatra
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        exclude:
+          - os: windows-latest
+            gem: ethon
```

Comment:
Are there comments we could add (now or in followup PRs) about why these tests need / ought to be excluded for these OSes?